libcontainer/container_linux.go

```diff
@@ -105,8 +105,8 @@ func (c *linuxContainer) Start(process *Process) error {
 		return newSystemError(err)
 	}
 	if err := parent.start(); err != nil {
-		// terminate the process to ensure that it properly is reaped.
-		if err := parent.terminate(); err != nil {
+		// wait for the process to ensure that it properly is reaped.
+		if _, err := parent.wait(); err != nil {
```

Comment:
yup, I'll have to look a bit more closely as I'm not all that familiar with this codebase yet in terms of providing a less naive fix.